include/cpuinfo_s390x.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 IBM
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CPU_FEATURES_INCLUDE_CPUINFO_S390X_H_
+#define CPU_FEATURES_INCLUDE_CPUINFO_S390X_H_
+
+#include "cpu_features_cache_info.h"
+#include "cpu_features_macros.h"
+
+CPU_FEATURES_START_CPP_NAMESPACE
+
+typedef struct {
+  int esan3: 1;
+  int zarch: 1;
+  int stfle: 1;
+  int msa: 1;
+  int ldisp: 1;
+  int eimm: 1;
+  int dfp: 1;
+  int edat: 1;
+  int etf3eh: 1;
+  int highgprs: 1;
+  int te: 1;
+  int vx: 1;
+  int vxd: 1;
+  int vxe: 1;
+  int gs: 1;
+  int vxe2: 1;
+  int vxp: 1;
+  int sort: 1;
+  int dflt: 1;
+  int vxp2: 1;
+  int nnpa: 1;
+  int pcimio: 1;
+  int sie: 1;
+
+  // Make sure to update S390XFeaturesEnum below if you add a field here.
+} S390XFeatures;
+
+typedef struct {
+  S390XFeatures features;
+} S390XInfo;
+
+S390XInfo GetS390XInfo(void);
+
+typedef struct {
+  char platform[64];       // 0 terminated string
+} S390XPlatformTypeStrings;
+
+typedef struct {
+  char num_processors[64];       // 0 terminated string
```

Comment:
What's the format of this string?

Reply:
This is just the line in /proc/cpuinfo that shows number of processors in form "# processors    : <processors>"

Reply:
Hmm it's probably best to get this field in a usable form.
I would make it an `int num_processors; // -1 if N/A`.

Reply:
Ah right, that makes sense